miniapp/miniapp.cpp

```diff
@@ -202,11 +198,7 @@ void banner() {
     std::cout << "  starting miniapp\n";
     std::cout << "  - " << threading::description() << " threading support\n";
     std::cout << "  - communication policy: " << std::to_string(global_policy::kind()) << " (" << global_policy::size() << ")\n";
-#ifdef NMC_HAVE_CUDA
-    std::cout << "  - gpu support: on\n";
-#else
-    std::cout << "  - gpu support: off\n";
-#endif
+    std::cout << "  - gpu support: " << (config::has_cuda? "on": "off") << "\n";
```

Comment:
The parameter should stay where it is in `nest::mc::config`.
I think that its use is appropriate here, because we are printing to the output whether the miniapp has gpu support, which it does by virtue of being compiled with the CUDA back end.

The `config::has_cuda` value is also used below to choose at run time whether to select the multicore or gpu back end for the model. This was not an option before, and it could be made a runtime choice. But I am inclined to make the rule `by default miniapp is multicore, and if you compile with CUDA it uses the GPU`, because that is what we had before.